packages/start-plugin/src/index.ts

```diff
@@ -0,0 +1,85 @@
+import { fileURLToPath, pathToFileURL } from 'node:url'
+
+import { compileStartOutput } from './compilers'
+
+import { logDiff } from './logger'
+import type { Plugin } from 'vite'
+
+const debug =
+  process.env.TSR_VITE_DEBUG &&
+  ['true', 'start-plugin'].includes(process.env.TSR_VITE_DEBUG)
+
+export type TanStackStartViteOptions = {
+  env: 'server' | 'ssr' | 'client'
+}
+
+const transformFuncs = [
+  'createServerFn',
+  'createMiddleware',
+  'serverOnly',
+  'clientOnly',
+  'createIsomorphicFn',
+]
+const tokenRegex = new RegExp(transformFuncs.join('|'))
+// const eitherFuncRegex = new RegExp(
+//   `(function ${transformFuncs.join('|function ')})`,
+// )
+
+export function TanStackStartVitePlugin(
+  opts: TanStackStartViteOptions,
+): Plugin {
+  let ROOT: string = process.cwd()
+
+  return {
+    name: 'vite-plugin-tanstack-start-create-server-fn',
+    enforce: 'pre',
+    configResolved: (config) => {
+      ROOT = config.root
+    },
+    transform(code, id) {
+      const url = pathToFileURL(id)
+      url.searchParams.delete('v')
+      id = fileURLToPath(url).replace(/\\/g, '/')
+
+      const includesToken = tokenRegex.test(code)
+      // const includesEitherFunc = eitherFuncRegex.test(code)
+
+      if (
+        !includesToken
+        // includesEitherFunc
+        // /node_modules/.test(id)
+      ) {
+        return null
+      }
+
+      if (code.includes('@react-refresh')) {
```

Comment:
Do we still need this?

Reply:
Yeah. It ensures that the plugins are in the right order.